public/pages/DetectorResults/containers/AnomalyResultsTable.tsx

```diff
@@ -62,6 +67,99 @@ export function AnomalyResultsTable(props: AnomalyResultsTableProps) {
       : [];
 
+  const handleOpenDiscover = async (startTime: number, endTime: number, item: any) => {
+    try {
+      // calculate time range with 10-minute buffer on each side per customer request
+      const TEN_MINUTES_IN_MS = 10 * 60 * 1000;
+      const startISO = new Date(startTime - TEN_MINUTES_IN_MS).toISOString();
+      const endISO = new Date(endTime + TEN_MINUTES_IN_MS).toISOString();
+      
+      const basePath = `${window.location.origin}${window.location.pathname.split('/app/')[0]}`;
+      
+      const savedObjectsClient = getSavedObjectsClient();
+      
+      const indexPatternTitle = props.detectorIndex.join(',');
+      
+      // try to find an existing index pattern with this title
+      const indexPatternResponse = await savedObjectsClient.find({
+        type: 'index-pattern',
+        fields: ['title'],
+        search: `"${indexPatternTitle}"`,
+        searchFields: ['title'],
+      });
+      
+      let indexPatternId;
+      
+      if (indexPatternResponse.savedObjects.length > 0) {
+        indexPatternId = indexPatternResponse.savedObjects[0].id;
+      } else {
+        // try to create a new index pattern
+        try {
+          const newIndexPattern = await savedObjectsClient.create('index-pattern', {
+            title: indexPatternTitle,
+            timeFieldName: props.detectorTimeField,
+          });
+          
+          indexPatternId = newIndexPattern.id;
+
+          getNotifications().toasts.addSuccess(`Created new index pattern: ${indexPatternTitle}`);
+        } catch (error) {
+          getNotifications().toasts.addDanger(`Failed to create index pattern: ${error.message}`);
+          return;
+        }
+      }
+      
+      // put query params for HC detector
+      let queryParams = '';
+      if (props.isHCDetector && item[ENTITY_VALUE_FIELD]) {
+        const entityValues = item[ENTITY_VALUE_FIELD].split('\n').map((s: string) => s.trim()).filter(Boolean);
+        const filters = entityValues.map((entityValue: string) => {
+          const [field, value] = entityValue.split(': ').map((s: string) => s.trim());
+          return `('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'${indexPatternId}',key:${field},negate:!f,params:(query:${value}),type:phrase),query:(match_phrase:(${field}:${value})))`;
+        });
+        
+        queryParams = `filters:!(${filters.join(',')}),`;
+      }
+
+      const discoverUrl = `${basePath}/app/data-explorer/discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(indexPattern:'${indexPatternId}',view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'${startISO}',to:'${endISO}'))&_q=(${queryParams}query:(language:kuery,query:''))`;
+      
+      window.open(discoverUrl, '_blank');
+    } catch (error) {
+      getNotifications().toasts.addDanger('Error opening discover view');
+    }
+  };
+
+  const getCustomColumns = () => {
```

Comment:
not sure what these are used for so please ignore if this doesn't make sense. but you can see in the discover url there is a columns too. so technically can set the columns as well. if we wwanted to carry over some state information. but like i said if this a different type of columns then pls ignore

Reply:
Yeah it has nothing to do with the discover url. Different detectors on different pages will have different columns in this table. The column customization logic used to be scattered in various places, but in this PR, I consolidated all of it into this method :)